### DISCUSSION TITLE
Enhancements to Question Editor and Question Question set service

### DISCUSSION THREAD

**Discussion by Prateek-slokam** [Sunbird-inQuiry/Community#56] [General Discussions]
This discussion is for the community for a look into the various enhancements that have been taken in through the use case of observation and survey as the asset types under the object of the question and question set. 
The observation and survey required 
1. Addition to the current question types like slider or text and validations like number only, character limits, attachment required etc.
2. Defining parent child relationship among questions
3. Enhancement to tree structure for the inclusion of bringing in taxonomy structures
4. Nested grouping structure for the questions (question sets within question sets)
5. Flexible scoring definition and min max scores across different sets
6. Multi grouping of questions (same question over different grouping) - like criteria based and evidence collection method (ECM) based in case of observation

You can further go into the details of these developments and enhancements through the following documents and stories
PRD - https://project-sunbird.atlassian.net/l/cp/Fqa7UZp1
Epics on Jira:
- https://project-sunbird.atlassian.net/browse/IQ-137
- https://project-sunbird.atlassian.net/browse/IQ-128

Of these, most of the enhancements as laid out are already developed over the inQuiry BB, the enhancements towards observation with rubric and preview flow remains.

@aks30 @vijiurs @kiranharidas187 @vrayulu

**Comment by kiranharidas187** [comment]
As mentioned by Prateek Observation without rubrics and  Survey  implementation has been done already. Please find the design doc for the same https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/2846064690/Survey+and+Observation+-+New+Question+Types

 @vrayulu Following is the design doc for observation with rubrics part. Please let us know your feedbacks.
https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/2983002141/Design+-+Observation+with+Rubrics

**Reply by kiranharidas187** [reply]
@vrayulu This is the latest doc for observation with rubrics. Please ignore the previous one and refer to this
https://project-sunbird.atlassian.net/wiki/spaces/~62428bf0699649006ae5562d/pages/3242721314/Changes+to+the+Question+Set+Schema+and+Primary+Category+of+Observation+with+rubrics+for+the+implementation+of+Observation+with+Rubric+creation

**Reply by kiranharidas187** [reply]
@vrayulu the same document has been updated based on our previous call discussion. **groupingAttribute** has been added  to the category definition section

**Comment by surendrasinghs** [comment]
@Prateek-slokam @kiranharidas187 I have commented on the various links above. I also noticed many comments provided long ago still unresolved on the PRD and a few other places. Many/most of the linked documents/links are also outdated. Feeling a bit lost… Please share the latest scope and plan for this so that I/others can engage with the relevant items.

@Krishnaj20 @vinukumar-vs Please check out coKreat related workflows in the PRD and tag Mani as well. thanks!
